modules/imgproc/src/smooth.cpp

```diff
@@ -2360,7 +2360,7 @@ void cv::GaussianBlur( InputArray _src, OutputArray _dst, Size ksize,
                _src.rows() > ksize.height && _src.cols() > ksize.width);
     (void)useOpenCL;
 
-    CV_IPP_RUN(!useOpenCL, ipp_GaussianBlur( _src,  _dst,  ksize, sigma1,  sigma2, borderType));
+    CV_IPP_RUN(!useOpenCL && _src.getObj() != _dst.getObj(), ipp_GaussianBlur( _src,  _dst,  ksize, sigma1,  sigma2, borderType));
```

Comment:
This condition is not accurate.
We should compare values of `cv::Mat::data` fields (in `ipp_GaussianBlur` function to hide implementation details).
```
Mat a = ...
Mat b = a;
GaussianBlur(a, b , ...); // .getObj() is different, but `.data` is the same.
```

Perhaps OpenCL path may not work properly for inplace calls too.